File: 0231-power-of-two/0231-power-of-two.cpp
class Solution {
public:
    bool isPowerOfTwo(int n) {
        int d=n/2;
        int r=n%2;
        while (d>0 && r==0){
            r=d%2;
            d=d/2;
        }
        if (d==0 && r==1){
            return true;
        }
        return false;
    }
};

File: 0231-power-of-two/0231-power-of-two.py
"""
계속 2로 나누다가 몫 0, 나머지 1이면 return True

"""

class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        d,r=divmod(n,2)
        while d>0 and r==0:
            d,r=divmod(d,2)
        if d==0 and r==1:
            return True
        return False